src/npctalk.cpp

```diff
@@ -2293,6 +2343,13 @@ void talk_effect_t::parse_sub_effect( JsonObject jo )
         subeffect_fun.set_npc_cbm_recharge_rule( setting );
     } else if( jo.has_member( "mapgen_update" ) ) {
         subeffect_fun.set_mapgen_update( jo, "mapgen_update" );
+    } else if( jo.has_string( "u_buy_monster" ) ) {
+        const std::string &monster_type_id = jo.get_string( "u_buy_monster" );
+        const int cost = jo.get_int( "cost", 0 );
+        const int count = jo.get_int( "count", 1 );
+        const bool pacified = jo.get_bool( "pacified", false );
+        const std::string &name = jo.get_string( "name", "" );
```

Comment:
This probably needs to be translated. I suggest using class `translation` so it will properly update after switching language. You might also want to update the string extraction script so it can actually be translated.

Reply:
That doesn't seem unreasonable. 

However, if I'm following correctly how `translation` is to be used, it seems like we need to push that change all the way down to the `unique_name` member in [the monster class](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/monster.h#L440) to actually accomplish the "properly update after switching language"--all my code is doing is setting that value, as is done in e.g. mapgen place_monster or when renaming a pet.

Given that, I think it's better tackled in a separate PR, because we'd need to review and/or update all those usages, which would include user entered text when renaming the pet, serializing/deserializing the unique name of the monster in the save, generating unique named corpses, and spawning them.

Reply:
I mean, it doesn't need to be that complicated. You're reading a name from json here, which can be translated (So your Roach will be recognized by Geralts from all over the world). I think it only needs changing `const std::string &name = jo.get_string( "name", "" );` to `translation name; jo.read( "name", name );`, and use `translation` instead of `std::string` for `name` in `talk_effect_fun_t::set_u_buy_monster` accordingly. `unique_name` does not need to be translated, as it is supposed to be modified by the player.

Reply:
Ok, that makes sense. I think the comment about it properly updating after switching languages tripped me up, because doing what you describe here and leaving `unique_name` as a string means that if you trigger the effect while in English and get a horse named Roach, when you switch to Polish it's going to continue to be called Roach instead of Płotka.